corehq/apps/data_interfaces/tasks.py

```diff
@@ -220,37 +214,27 @@ def task_generate_ids_and_operate_on_payloads(
     repeater_id: Optional[str],
     domain: str,
     action,  # type: Literal['resend', 'cancel', 'requeue']  # 3.8+
-    use_sql: bool,
+    use_sql: bool = True,
 ) -> dict:
-    repeat_record_ids = _get_repeat_record_ids(payload_id, repeater_id, domain,
-                                               use_sql)
-    return operate_on_payloads(repeat_record_ids, domain, action, use_sql,
+    repeat_record_ids = _get_repeat_record_ids(payload_id, repeater_id, domain)
+    return operate_on_payloads(repeat_record_ids, domain, action,
                                task=task_generate_ids_and_operate_on_payloads)
 
 
-def _get_repeat_record_ids(
-    payload_id: Optional[str],
-    repeater_id: Optional[str],
-    domain: str,
-    use_sql: bool,
-) -> List[str]:
-    if not payload_id and not repeater_id:
-        return []
+def _get_repeat_record_ids(payload_id, repeater_id, domain):
     if payload_id:
-        if use_sql:
-            records = get_sql_repeat_records_by_payload_id(domain, payload_id)
-            return [r.id for r in records]
-        else:
-            return get_couch_repeat_record_ids_by_payload_id(domain, payload_id)
+        queryset = SQLRepeatRecord.objects.filter(
+            domain=domain,
+            payload_id=payload_id,
+        )
+    elif repeater_id:
+        queryset = SQLRepeatRecord.objects.filter(
+            domain=domain,
+            repeater__id=repeater_id,
+        )
     else:
-        if use_sql:
-            queryset = SQLRepeatRecord.objects.filter(
-                domain=domain,
-                repeater__id=repeater_id,
-            )
-            return [r['id'] for r in queryset.values('id')]
-        else:
-            return list(iter_repeat_record_ids_by_repeater(domain, repeater_id))
+        return []
```

Comment:
@millerdev was the change to return an empty list if neither a `payload_id` or `repeater_id` were specified intentional? I just ran into this issue where I wanted to cancel all pending repeat records for a domain in the UI (so not specifying either a payload_id or repeater), and the celery task never ran since there weren't any repeat records returned.

Reply:
I think this is a bug, and it should be
```py
queryset = SQLRepeatRecord.objects.filter(domain=domain)
```

Reply:
On further investigation I'm not so sure it's a bug. We have quite a few tests, including [these](https://github.com/dimagi/commcare-hq/blob/923453dae7631d839baf15528461b77a79f5e77d/corehq/apps/data_interfaces/tests/test_utils.py#L560-L568), that require an empty result set for this conditional branch. I need to think about this some more, but I'm wondering if the reason for this is because it would potentially return a huge result set for some domains, and introducing pagination here would require significant rework?